kuma/javascript/src/task-completion-survey.jsx

```diff
@@ -0,0 +1,177 @@
+// @flow
+import * as React from 'react';
+import { useContext, useEffect, useState } from 'react';
+import css from '@emotion/css';
+
+import CloseIcon from './icons/close.svg';
+import DocumentProvider from './document-provider.jsx';
+import GAProvider from './ga-provider.jsx';
+import LocaleProvider from './locale-provider.jsx';
+import UserProvider from './user-provider.jsx';
+import gettext from './gettext.js';
+
+const styles = {
+    notification: css({
+        position: 'fixed',
+        display: 'flex',
+        flexDirection: 'row',
+        top: 15,
+        right: 15,
+        width: 300,
+        padding: 15,
+        paddingRight: 8,
+        zIndex: 999999,
+        fontSize: 16,
+        backgroundColor: '#e4f0f5',
+        color: '#333',
+        border: 'solid #3d7e9a 2px',
+        borderRadius: 2,
+        boxShadow: '3px 3px 5px rgba(0, 0, 0, 0.25)'
+    }),
+    dismissButton: css({
+        alignSelf: 'start',
+        padding: 0,
+        borderWidth: 0,
+        backgroundColor: 'inherit'
+    }),
+    dismissIcon: css({
+        height: 18
+    })
+};
+
+// This is the name of the waffle flag that controls this notification
+const WAFFLE_FLAG = 'sg_task_completion';
+
+// This is the name of a localStorage property we use to prevent the
+// display of the notification to someone who has dismissed it or
+// responded to it within the last month
+const EMBARGO_TIMESTAMP = 'taskTracker';
+
+// This is how long the embargo lasts
+const EMBARGO_LENGTH = 1000 * 60 * 60 * 24 * 32; // 32 days
+
+// If the user dismisses or responds to the survey, then we call this
+// function to set the embargo timestamp. During this embargo period,
+// the notification will not be shown again to the user.
+function embargo() {
+    localStorage.setItem(
+        EMBARGO_TIMESTAMP,
+        String(Date.now() + EMBARGO_LENGTH)
```

Comment:
Is the `String()` really necessary? E.g.

```
> sessionStorage.foo = "4"
"4"
> sessionStorage.foo > 3
true
> sessionStorage.foo > 5
false
```

The browsers will force everything to a string anyway and JS will automatically cast strings to ints when you use the `>` operator?

Not trying to change anything, just curious to learn.

Reply:
String() is probably not necessary, but:

- flow might complain if I don't use it (I haven't checked, though)
- the original specs for localStorage said that they'd allow non-string values and serialize them in the way that things like postMessage() do. Browsers never implemented that, and maybe the specs have now changed to say that it is string only, but I'm still pissy about that. Forcing to String means that things won't break if the implementation later starts accepting other types
- using String() here helps to make it clear why we're using parseInt() when we later read the data out of localStorage